### ISSUE TITLE
Add Parametric Lap joint

### ISSUE THREAD

**Issue by yck011522** [gramaziokohler/integral_timber_joints#102]
As discussed with Davide, we will implement a lap joint that is controlled by 8 control points. 
The setting out of the joint is similar to the NP joint.

We will keep a list of the corner intersection point in the joint attributes, including a centre point (plane).
Points will be kept in WCF. 
Side and mid planes (3 on each side) can be derived from these points and center plane.
The control points will be derived from an offset from these side planes in a 2.5D manner.
Each control point have 2 unique parameters (16 in total) and share the depth parameter.

**Comment by yck011522** [comment]
As furthered explored with Davide, the previous expectation of having 8 points are not enough to deal with many situations.
One such example is that for some joints, 8 points cannot be used to describe a rotated (relationship of the two-beam changed) but is a geometrical identical joint. See the crossed out one:

![2021-09-29 15_30_30](https://user-images.githubusercontent.com/1297653/135278352-c106ff8b-13ab-43be-9e77-08263d05027a.jpg)

The conclusion is that it is easier to manage a joint with 4 polylines drawn. Each of them describing two sides of the lap joint. The two sides are joined at a specific height.

**Comment by yck011522** [comment]
Some strings for testing:

```
[50.,[[[0.,0.],[0.,0.2],[1.,0.2],[1.,0.],],[[0.,0.],[1.,0.],],[[0.,0.],[0.,0.2],[1.,0.2],[1.,0.],],[[0.,0.],[1.,0.],],]]

[50.,[[[0.,0.],[0.2,0.],[0.2,0.2],[0.8,0.2],[0.8,0.],[1.,0.],],[[0.,0.],[0.,0.2],[1.,0.2],[1.,0.],],[[0.,0.],[0.2,0.],[0.2,0.2],[0.8,0.2],[0.8,0.],[1.,0.],],[[0.,0.],[0.,0.2],[1.,0.2],[1.,0.],],]]

[50.,[[[0.,0.],[0.038,0.192],[1.,0.],],[[0.,0.],[1.,0.],],[[0.,0.],[0.963,0.193],[1.,0.],],[[0.,0.],[0.193,0.037],[0.808,0.038],[1.,0.],],]]

[50.,[[[0.,0.],[1.,0.],],[[0.,0.],[0.5,0.2],[1.,0.],],[[0.,0.],[1.,0.],],[[0.,0.],[0.5,0.2],[1.,0.],],]]

[50.,[[[0.,0.],[0.2,0.2],[0.8,0.2],[1.,0.],],[[0.,0.],[1.,0.],],[[0.,0.],[0.2,0.2],[0.8,0.2],[1.,0.],],[[0.,0.],[1.,0.],],]]
```

**Comment by yck011522** [comment]
closed via e00fb8b496a3b2be505e5443b6f5063246dcaf09
